File: Exercise/dice-game/java/Dice.java
import java.util.Random;

public class Dice {
        private FaceValue faceValue = FaceValue.One;
        
        public void roll() {
                faceValue = FaceValue.values()[new Random().nextInt(FaceValue.values().length)];
        }

        public FaceValue getFaceValue() {
                return faceValue;
        }
}

File: Exercise/dice-game/c/dice.c
#include "common.h"
#include "dice.h"

void dice_initialize(dice_t *dice) {
        ASSERT_NOT_NULL(dice);

        dice->face_value = one;
}

void dice_finalize(dice_t *dice) {
        ASSERT_NOT_NULL(dice);
        // NULL
}

void dice_roll(dice_t *dice) {
        ASSERT_NOT_NULL(dice);

        dice->face_value = (face_value_t) (rand() % 6) + 1;
}

face_value_t dice_face_value(dice_t *dice) {
        ASSERT_NOT_NULL(dice);

        return dice->face_value;
}
